x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -235,31 +240,47 @@ public TokenService(Settings settings, Clock clock, Client client, XPackLicenseS
     }
 
     /**
-     * Creates an access token and optionally a refresh token as well, based on the provided authentication and metadata with an
-     * auto-generated token document id. The created tokens are stored in the security index.
+     * Creates an access token and optionally a refresh token as well, based on the provided authentication and metadata with
+     * auto-generated values. The created tokens are stored in the security index for versions up to 7.1.0 and to
+     * a specific security tokens index for later versions.
      */
     public void createOAuth2Tokens(Authentication authentication, Authentication originatingClientAuth,
                                    Map<String, Object> metadata, boolean includeRefreshToken,
-                                   ActionListener<Tuple<UserToken, String>> listener) {
+                                   ActionListener<Tuple<String, String>> listener) {
```

Comment:
This was not required in the context of this PR, but it bothered me for a while because we would call the listener with the UserToken object, when all we need is the access token to return to the user.

Reply:
Thanks for tackling this.